sqlalchemy_ibmi/base.py

```diff
@@ -811,6 +812,13 @@ def create_connect_args(self, url):
                 opts['DefaultLibraries'] += ','.join(
                     opts.pop('library_list', ''))
 
+        if 'trim_char_fields' in opts:
+            try:
+                opts['TrimCharFields'] = str(util.strtobool(opts['trim_char_fields']))
+                opts.pop('trim_char_fields', '')
+            except (ValueError, KeyError):
+                opts.pop('trim_char_fields', '')
+
```

Comment:
KeyError is not actually needed here, since we've already tested that the key exists in the `if` check above. We can also reduce the redundant pop in the exception handler by doing that first, since we're always going to want to remove it if it's there. In addition, we can reduce a level of nesting by removing the if check (since we need to handle KeyError regardless, otherwise I would have preferred to leave the if and remove the exception handling)

```suggestion
            try:
                trim_char_fields = opts.pop('trim_char_fields')
                opts['TrimCharFields'] = str(util.strtobool(trim_char_fields))
            except (ValueError, KeyError):
                pass
```

Reply:
I understand your suggestion.  The single pop makes sense. 

Question: Should ValueError and KeyError be separate exceptions?  If the 'trim_char_fields' is not there then continuing on is ok.  But if I specified a bad value for it then wouldn't we want to throw the exception?

Reply:
Currently we don't do that for `use_system_naming`, but maybe we should.

Also would be good to move this addition below that block, since they're very similar in behavior.

Reply:
I took your proposed block in and moved it as suggested.  Also split out the ValueErrors for both use_system_naming & trim_char_fields.